### DISCUSSION TITLE
Creation of API tokens

### DISCUSSION THREAD

**Discussion by andre-pt** [lae/ansible-role-proxmox#218] [Q&A] [answered]
Hi

Is there a way to create API tokens which is not documented?
If not, what would be the recommended way? using shell commands?

Thanks a lot
Andre

**Comment by lae** [comment, marked as answer]
Having checked the PVE API documentation (https://pve.proxmox.com/pve-docs/api-viewer/index.html#/access/users/{userid}/token/{tokenid}) it looks like token management has its own configuration endpoint, and it's not touched at all in this role. So, there isn't any way to create API tokens with this role, at least at the moment.

I don't have a recommended way, but shell commands using `pvesh` would probably work. You would just run them on one host in the cluster.

**Reply by andre-pt** [reply]
thanks for the reply! I will try to do it over shell.
